test/decl/typealias/typealias.swift

```diff
@@ -135,13 +135,55 @@ protocol CB {
   func setIt(element: E)
 }
 
-func go1<T : CB, U : Col where U.Elem == T.E>(col: U, builder: T) {
+func go1<T : CB, U : Col where U.Elem == T.E>(col: U, builder: T) { // OK
   builder.setIt(col.elem)
 }
 func go2<T : CB, U : Col where U.Elem == T.C.Elem>(col: U, builder: T) { // OK
   builder.setIt(col.elem)
 }
 
+// Test for same type requirement with typealias == concrete
+func go3<T : CB where T.E == Int>(builder: T) {
+  builder.setIt(1)
+}
+
+// Test for conformance to protocol with associatedtype and another with typealias with same name.
+protocol Iterator {
```

Comment:
To avoid confusion when debugging tests, we usually try to avoid shadowing standard library names.  For example, we use a My~ prefix.